apollo-router-core/src/spec/field_type.rs

```diff
@@ -0,0 +1,106 @@
+use crate::*;
+use apollo_parser::ast;
+
+// Primitives are taken from scalars: https://spec.graphql.org/draft/#sec-Scalars
+#[derive(Debug)]
+pub(crate) enum FieldType {
+    Named(String),
+    List(Box<FieldType>),
+    NonNull(Box<FieldType>),
+    String,
+    Int,
+    Float,
+    Id,
+    Boolean,
+}
+
+impl FieldType {
+    pub(crate) fn validate_value(&self, value: &Value, schema: &Schema) -> bool {
```

Comment:
is_valid? I d expect validate_value to return Result<()>

Reply:
Very good point :sweat: naming's hard

Reply:
SO! As discussed I'm not too much into renaming to a method prefixed with `is_` because they are usually low-cost methods that consult something. I decided to replace the bool with results. It's a bit les mumbo jumbo with the bool expression but a bit more into collecting Result.

Let me know if you don't like either and I will just rename validate_value is is_value_valid

Fixed in 10e3e17